hera_qm/firstcal_metrics.py

```diff
@@ -0,0 +1,542 @@
+            firstcal delay solutions in seconds
+
+        self.delay_offsets : ndarray, shape=(N_ant, N_times)
+            firstcal delay solution offsets from time average
+
+        self.frac_JD : ndarray, shape=(N_times,)
+            ndarray containing time-stamps of each integration
+            in units of the fraction of current JD
+            i.e. 2457966.53433 -> 0.53433
+
+        self.ants : ndarray, shape=(N_ants,)
+            ndarray containing antenna numbers
+        """
+
+        # Instantiate UVCal and read calfits
+        self.UVC = UVCal()
+        self.UVC.read_calfits(calfits_file)
+
+        # get file prefix
+        self.file_stem = '.'.join(calfits_file.split('.')[:-1])
+
+        # Calculate median delay
+        self.delays = self.UVC.delay_array.squeeze() * 1e9
+        self.delay_avgs = np.median(self.delays, axis=1)
+        self.delay_offsets = (self.delays.T - self.delay_avgs).T
+
+        # get other relevant arrays
+        self.times = self.UVC.time_array
+        self.start_JD = np.floor(self.times).min()
+        self.frac_JD = self.times - self.start_JD
+        self.minutes = 24 * 60 * (self.frac_JD - self.frac_JD.min())
+        self.Nants = self.UVC.Nants_data
+        self.ants = self.UVC.ant_array
+        self.version_str = hera_qm_version_str
+
+    def run_metrics(self, std_cut=0.5, output=False):
+        """
+        Run all metrics and attach to class
+
+        filename : str, default=None
+            filename for output w/o filetype suffix
+            if None, uses calfile stem
+
+        std_cut : float, default=0.5
+            delay stand. dev cut for good / bad determination
+
+        output : bool, default=False
+            return with function
+
+        Output:
+        -------
+        Create a self.metrics dictionary
+        if output == True:
+            return self.metrics
+
+        """
+        # Calculate std and zscores
+        (self.ant_avg, self.ant_std, self.time_std, self.agg_std,
+         self.z_scores) = self.delay_std(return_dict=True)
+
+        # Given delay std cut find "bad" ant
+        # determine if full sol is bad
+        if self.agg_std > std_cut:
+            self.full_sol = 'bad'
+        else:
+            self.full_sol = 'good'
+
+        self.bad_ants = []
+        for ant in self.ant_std:
+            if self.ant_std[ant] > std_cut:
+                self.bad_ants.append(ant)
+
+        # put into dictionary
+        metrics = OrderedDict()
+        metrics['full_sol'] = self.full_sol
+        metrics['bad_ant'] = self.bad_ants
+        metrics['z_scores'] = self.z_scores
+        metrics['ant_avg'] = self.ant_avg
+        metrics['ant_std'] = self.ant_std
+        metrics['time_std'] = self.time_std
+        metrics['agg_std'] = self.agg_std
+        metrics['times'] = self.times
+        metrics['version'] = self.version_str
+        self.metrics = metrics
+
+        if output is True:
+            return self.metrics
```

Comment:
Alright, then I think that my recommendation is to get rid of this. I think it's a little confusing anyway for a function to return a value only if asked to, and the name (`run_metrics`) implies that it just computes the metrics (as opposed to, say `get_metrics`). But I think the name should stay as is, since it's nice and straightforward.